docs/src/reference/asciidoc/core/kerberos.adoc

```diff
@@ -0,0 +1,484 @@
+that is on the classpath when the Spark application is submitted can offer implementations to be loaded and used.
+`EsServiceCredentialProvider` is one such implementation that is loaded whenever {eh} is on the job's classpath.
+
+Once loaded, `EsServiceCredentialProvider` determines if {krb} authentication is enabled for {eh}. If it is determined
+that {krb} authentication is enabled for {eh}, then the credential provider will automatically obtain delegation tokens
+from {es} and add them to the credentials on the YARN application submission context. Additionally, in the case that
+the job is a long lived process like a Spark Streaming job, the credential provider is used to update or obtain new
+delegation tokens when the current tokens approach their expiration time.
+
+The time that Spark's credential providers are loaded and called depends on the cluster deploy mode when submitting your
+Spark app. When running in `client` deploy mode, Spark runs the user's driver code in the local JVM, and launches the
+YARN application to oversee the processing as needed. The providers are loaded and run whenever the YARN application
+first comes online. When running in `cluster` deploy mode, Spark launches the YARN application immediately, and the
+user's driver code is run from the resulting Application Master in YARN. The providers are loaded and run _immediately_,
+before any user code is executed.
+
+[[kerberos-spark-credprovider-conf]]
+===== Configuring the credential provider
+
+All implementations of the Spark credential providers use settings from only a few places:
+
+1. The entries from the local Hadoop configuration files
+2. The entries of the local Spark configuration file
+3. The entries that are specified from the command line when the job is initially launched
+
+Settings that are configured from the user code are not used because the provider must run once for all jobs that are
+submitted for a particular Spark application. User code is not guaranteed to be run before the provider is loaded.
+To make things more complicated, a credential provider is only given the local Hadoop configuration to determine if
+they should load delegation tokens.
+
+These limitations mean that the settings to configure {eh} for {krb} authentication need to be in specific places:
+
+First, `es.security.authentication` MUST be set in the local Hadoop configuration files as 'kerberos'. If it is not set
+in the Hadoop configurations, then the credential provider will assume that 'simple' authentication is to be used, and
+will not obtain delegation tokens.
+
+Secondly, all general connection settings for {eh} (like `es.nodes`, `es.ssl.enabled`, etc...) must be specified either
+in the local Hadoop configuration files, in the local Spark configuration file, or from the command line. If these
+settings are not available here, then the credential provider will not be able to contact {es} in order to obtain the
+delegation tokens that it requires.
```

Comment:
Similar comment as above about requiring connection information in core-site.xml , perhaps a section with an example that we can link ? (I didn't see any other example in our docs for configuration from there).